query/src/test/java/com/connexta/replication/query/QueryServiceToolsTest.java

```diff
@@ -0,0 +1,128 @@
+import com.connexta.replication.api.persistence.FilterManager;
+import com.connexta.replication.api.persistence.SiteManager;
+import com.connexta.replication.api.queue.QueueBroker;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QueryServiceToolsTest {
+
+  @Mock private NodeAdapters nodeAdapterFactories;
+
+  @Mock private SiteManager siteManager;
+
+  @Mock private FilterManager filterManager;
+
+  @Mock private FilterIndexManager filterIndexManager;
+
+  @Mock private QueueBroker queueBroker;
+
+  private long period = 60L;
+
+  private QueryServiceTools queryServiceTools;
+
+  @Before
+  public void setUp() throws Exception {
+    queryServiceTools =
+        new QueryServiceTools(
+            nodeAdapterFactories,
+            siteManager,
+            filterManager,
+            filterIndexManager,
+            queueBroker,
+            period);
+  }
+
+  @Test
+  public void getPeriod() {
+    assertThat(queryServiceTools.getPeriod(), Matchers.is(period));
+  }
+
+  @Test
+  public void getPeriodDefault() {
+    queryServiceTools =
+        new QueryServiceTools(
+            nodeAdapterFactories, siteManager, filterManager, filterIndexManager, queueBroker, 0);
+    assertThat(queryServiceTools.getPeriod(), Matchers.is(TimeUnit.MINUTES.toSeconds(5)));
+  }
+
+  @Test
+  public void allSites() {
+    queryServiceTools.allSites();
+    verify(siteManager).objects();
+  }
+
+  @Test
+  public void getAdapterFor() {
+    Site site = mock(Site.class);
+    when(site.getType()).thenReturn(SiteType.DDF);
+    NodeAdapterFactory nodeAdapterFactory = mock(NodeAdapterFactory.class);
+    when(nodeAdapterFactories.factoryFor(any(SiteType.class))).thenReturn(nodeAdapterFactory);
+    queryServiceTools.getAdapterFor(site);
+    verify(nodeAdapterFactory).create(null);
+  }
+
+  @Test
+  public void activeFiltersFor() {
+    Filter filter = mock(Filter.class);
+    Site site = mock(Site.class);
+    when(filterManager.filtersForSite(null)).thenReturn(Stream.of(filter));
+    assertThat(queryServiceTools.activeFiltersFor(site).findFirst().get(), Matchers.is(filter));
+  }
+
+  @Test
+  public void getOrCreateFilterIndex() {
+    Filter filter = mock(Filter.class);
+    queryServiceTools.getOrCreateFilterIndex(filter);
+    verify(filterIndexManager).getOrCreate(filter);
+  }
+
+  @Test
+  public void saveFilterIndex() {
+    FilterIndex filterIndex = mock(FilterIndex.class);
+    queryServiceTools.saveFilterIndex(filterIndex);
+    verify(filterIndexManager).save(filterIndex);
+  }
+
+  @Test
+  public void getQueueFor() {
+    Site site = mock(Site.class);
+    when(site.getId()).thenReturn("siteid");
+    queryServiceTools.getQueueFor(site);
```

Comment:
❗️ There is no verification of the result here or the fact the queue broker is called with the id